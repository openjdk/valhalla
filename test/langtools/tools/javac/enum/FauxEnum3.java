/*
 * @test /nodynamiccopyright/
 * @bug 5009574
 * @summary verify an enum type can't be directly subclassed
 * @author Joseph D. Darcy
 *
 * @compile/fail/ref=FauxEnum3.out -XDrawDiagnostics  FauxEnum3.java
 * @compile/fail/ref=FauxEnum3.preview.out -XDrawDiagnostics --enable-preview -source ${jdk.version} FauxEnum3.java
 */

public class FauxEnum3 extends SpecializedEnum {
}

enum SpecializedEnum {
    RED {
        boolean special() {return true;}
    },
    GREEN,
    BLUE;
    boolean special() {return false;}
}
