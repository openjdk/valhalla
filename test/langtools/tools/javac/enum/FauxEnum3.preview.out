FauxEnum3.java:11:8: compiler.err.cant.inherit.from.sealed: SpecializedEnum
1 error